src/main/java/scala_maven/ScalaMojoSupport.java

```diff
@@ -783,47 +790,81 @@ private String getToolClasspath() throws Exception {
     return options;
   }
 
+  private String getScala3ArtifactId(String a) {
+    String _scalaVersionNStr = _scalaVersionN.toString();
+    // int indexOf_ = _scalaVersionNStr.indexOf("-");
+    return _scalaVersionN.major == 3 ? a + "_" + _scalaVersionNStr : a;
+  }
+
```

Comment:
I don't understand the purpose of this function and it's usage like
```
        _scalaVersionN.major == 3
            ? getScala3ArtifactId(SCALA3_LIBRARY_ARTIFACTID)
            : SCALA_LIBRARY_ARTIFACTID;
  ```

Reply:
I looked at scala3 on search.maven.org, I understand now the change, the naming convention has changed again (worst IMHO). So to support scala3 naming convention a refactor of the code should be done. The scala-maven-plugin should work in `incremental` mode (zinc) and `all` mode (scala compiler), both mode are impacted by thoses changes in naming.